_GNU_SOURCE must be defined before including stdio.h for asprintf().

--- src/lib/packet-show.c.orig	2014-02-17 19:39:10.097032026 +0100
+++ src/lib/packet-show.c	2014-02-17 19:39:20.887897122 +0100
@@ -52,6 +52,7 @@
  * Creates printable text strings from packet contents
  *
  */
+#define _GNU_SOURCE
 #include "config.h"
 
 #ifdef HAVE_SYS_CDEFS_H
